service/history/processing/interface.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//go:generate mockgen -copyright_file ../../../LICENSE -package $GOPACKAGE -source $GOFILE -destination interface_mock.go -self_package github.com/uber/cadence/service/history/processing
+
+package processing
```

Comment:
Any suggestion on the package name? This package will contain all the logic for managing tasks queues (loading, tracking/persisting ack level, sending task for processing etc.)

Reply:
Maybe [task|job][Manager|Controller]?

Reply:
How about queueManager? I think it makes sense since this package is largely responsible for creating and managing virtual queues.

I also don't like the Job word as it is too generic. What would you think about renaming it to something like VirtualQueue, TaskQueue or [Scoped|Filtered]Queue?

Reply:
Ideally I would like to use a single word for the package name, which is Golang convention and we follow that throughout the code base. So how about just using `queue` as the package name? As @emrahs mentioned, this package is largely responsible for creating and managing queues.

Reply:
Hmm, as a Golang newbie, I missed that convention. Found this write-up which has thorough recommendations about naming packages, and it's in line with @yycptt 's point about brevity: https://blog.golang.org/package-names

With that context, +1 from me for `queue`.